tests/nnstreamer_filter_snpe/unittest_filter_snpe.cc

```diff
@@ -0,0 +1,366 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * @file    unittest_filter_snpe.cc
+ * @date    16 Sep 2021
+ * @brief   Unit test for snpe tensor filter sub-plugin
+ * @author  Yongjoo Ahn <yongjoo1.ahn@samsung.com>
+ * @see     http://github.com/nnstreamer/nnstreamer
+ * @bug     No known bugs
+ *
+ */
+#include <gtest/gtest.h>
+#include <glib.h>
+#include <gst/gst.h>
+#include <unittest_util.h>
+
+#include <tensor_common.h>
+#include <nnstreamer_plugin_api_filter.h>
+#include <nnstreamer_util.h>
+
+/**
+ * @brief internal function to get model filename
+ */
+static void
+_GetModelFilePath (gchar ** model_file)
+{
+  const gchar *src_root = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  gchar *root_path = src_root ? g_strdup (src_root) : g_get_current_dir ();
+  gchar *model_name = g_strdup_printf ("add2_float.dlc");
+
+  *model_file = g_build_filename (
+      root_path, "tests", "test_models", "models", model_name, NULL);
+  g_free (model_name);
+  g_free (root_path);
+}
+
+/**
+ * @brief Set tensor filter properties
+ */
+static void
+_SetFilterProp (GstTensorFilterProperties *prop, const gchar *name, const gchar **models)
+{
+  memset (prop, 0, sizeof (GstTensorFilterProperties));
+  prop->fwname = name;
+  prop->fw_opened = 0;
+  prop->model_files = models;
+  prop->num_models = g_strv_length ((gchar **) models);
+}
+
+/**
+ * @brief Positive case with successful getModelInfo
+ */
+TEST (nnstreamerFilterSnpe, getModelInfo00)
+{
+  int ret;
+  void *data = NULL;
+  gchar *model_file;
+  GstTensorFilterProperties prop;
+  _GetModelFilePath (&model_file);
+  ASSERT_TRUE (g_file_test (model_file, G_FILE_TEST_EXISTS));
```

Comment:
how about adding this in _GetModelFilePath()
```
static gboolean
_GetModelFilePath (...)
{
  return g_file_test (model_file, G_FILE_TEST_EXISTS);
}
....
ASSERT_TRUE (_GetModelFilePath (&model_file));
```

Reply:
Thanks! Looks much better. I updated the code, PTAL